### PR TITLE
 uar-1412 add ch_reference field

### PR DESCRIPTION
**Pull request by lduranteau**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-1412

### Change description
add ch_reference field to trust in Mongo db

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesServiceTest.java [hunk 1]
```diff
@@ -116,10 +116,12 @@ private void testSubmissionCreation(boolean isResumeLinkExpected, String entityN
 
         var trustDataDto = new TrustDataDto();
         trustDataDto.setTrustName("dto");
+        trustDataDto.setTrustName("ch_reference - dto");
```
Comment (mwestacott) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesServiceTest.java:119]: This line overwrites the value set on the line above, so wondering what the reason for doing it is? Same on line 124.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/trust/TrustDataDto.java [hunk 2]
```diff
@@ -12,6 +12,7 @@ public class TrustDataDto {
 
     public static final String TRUST_ID_FIELD = "trust_id";
     public static final String TRUST_NAME_FIELD = "trust_name";
+    public static final String CH_REFERENCE = "ch_reference";
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/trust/TrustDataDto.java:15]: Please add "_FIELD" on to the end of the constant name.

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/mocks/Mocks.java [hunk 3]
```diff
@@ -331,6 +331,7 @@ private static List<TrustDataDto> buildTrustDataInFiling(int numberOfTrusts) {
         for (int i = 1; i <= numberOfTrusts; i++) {
             TrustDataDto trustDataDto = new TrustDataDto();
             trustDataDto.setTrustName("Trust Name " + i);
+            trustDataDto.setChReference("ch_reference" + i);
```
Comment (mwestacott) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/mocks/Mocks.java:334]: These mocks must be used by a unit test somewhere. Need to find which one(s) and update the assertions to check that the 'ch_reference' field is set correctly (I think).

Comment (lduranteau) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/mocks/Mocks.java:334]: I will add it

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesServiceTest.java (hunk 1 above)
Comment (lduranteau) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesServiceTest.java:119]: copy paste 🤦‍♂️

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/mocks/Mocks.java (hunk 3 above)
Comment (lduranteau) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/mocks/Mocks.java:334]: in reality, it is not necessary
I will remove it

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesServiceTest.java (hunk 1 above)
Comment (lduranteau) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesServiceTest.java:119]: in reality, it is not necessary
I will remove it

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/mocks/TrustMock.java [hunk 4]
```diff
@@ -10,6 +10,7 @@
 public class TrustMock {
     private static void populateCommonTrustDataDao(TrustDataDao trustDataDao) {
         trustDataDao.setTrustName("Trust Name");
+        trustDataDao.setChReference("ch_reference");
```
Comment (mwestacott) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/mocks/TrustMock.java:13]: This is really minor but I try to make test data values representative of 'real' possible data. Could you find out what kinds of 'ch_reference' values are stored in the web OE session and use those here (in the SDK PR too)?

**Review by mwestacott — APPROVED** [review]

**Pull request merged**